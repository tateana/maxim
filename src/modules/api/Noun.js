/* eslint no-underscore-dangle: 0 */  // --> OFF


export default class Noun {

    static genders = {
        'm': 'der',
        'f': 'die',
        'n': 'das'
    }

    constructor(de, ruOptions = null, gender, ru = null) {
        this._de = de

        this._ruOptions = ruOptions
        this._ru = ru || ruOptions[0]
        this._gender = gender
    }

    get created() {
        if (!this._created) {
            this._created = new Date()
        }

        return this._created
    }

    get origin() {
        return this._de
    }

    get de() {
        return this._de
    }

    get ru() {
        return this._ru
    }

    get gender() {
        return this._gender
    }

    get literalGender() {
        return Noun.genders[this._gender]
    }

    get translates() {
        return this._ruOptions || [this._ru]
    }

    get translate() {
        return this._ru
    }

    set translate(value) {
        this._ru = value
    }

    set gender(value) {
        this._gender = value
    }

    get isWritable() {
        return this._ruOptions && this._ruOptions.length > 0
    }


}