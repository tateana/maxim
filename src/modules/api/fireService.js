import firebase from 'firebase/app';
import "firebase/firestore";
import upperFirst from 'lodash.upperfirst';
import { from } from 'rxjs/index';


const config = {
    apiKey: process.env.REACT_APP_FIREBASE_KEY,
    authDomain: "maxim-6366b.firebaseapp.com",
    databaseURL: "https://maxim-6366b.firebaseio.com",
    projectId: "maxim-6366b",
    storageBucket: "maxim-6366b.appspot.com",
    messagingSenderId: "174473215042"
};

firebase.initializeApp(config);
const fireDb = firebase.firestore();
fireDb.settings({ timestampsInSnapshots: true })

export { fireDb }

export default class FireService {

    constructor(db, collectionName) {
        this.db = db;
        this.collection = this.db.collection(collectionName);
    }

    static docToEntity(doc) {
        throw new Error(`You have to implement convertion of ${typeof (doc)}  to an Entity`);
    }

    fetchAll(limit = 10) {
        return from(
            this.collection
                .limit(limit).get()
                .then((querySnapshot) => querySnapshot.docs.map(doc => this.constructor.docToEntity(doc)))
        )
    }

    fetchExact(id) {
        console.log('BADD')
        return from(
            this.collection.doc(upperFirst(id)).get()
                .then((doc) => {
                    if (doc.exists) {
                        return this.constructor.docToEntity(doc)
                    }

                    return null;
                })
        )
    }
}