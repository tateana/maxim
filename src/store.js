import { createStore } from 'redux'
import rootReducer from './modules/reducers'


const store = createStore(rootReducer)

export default store