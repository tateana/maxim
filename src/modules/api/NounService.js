import pick from 'lodash.pick';
import upperFirst from 'lodash.upperfirst';
import { from } from 'rxjs/index';
import FireService, { fireDb } from './fireService';
import Noun from './Entity/Noun';

class NounService extends FireService {

    static docToEntity(doc) {
        const data = doc.data();
        return new Noun(data.de, null, data.gender, data.ru)
    }

    static getName() {
        return 'words'
    }

    fetchLike(term, limit = 2) {
        const cleanedTerm = upperFirst(term)
        return from(
            this.collection
                .where("de", ">=", cleanedTerm).limit(limit).get()
                .then((querySnapshot) => querySnapshot.docs.filter(doc => doc.id.startsWith(cleanedTerm)).map(doc => {
                    const data = doc.data();
                    return new Noun(data.de, null, data.gender, data.ru)
                }))
        )
    }

    add(noun) {
        return from(this.collection
            .doc(noun.de)
            .set(pick(noun, ['de', 'ru', 'gender', 'created']))
            .then(() => true)
            // .catch(error => false)
        )
    }
}

export default new NounService(fireDb, 'nouns')