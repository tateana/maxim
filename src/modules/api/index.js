import Noun from './Entity/Noun';
import Score from './Entity/Score';

export { Noun, Score };