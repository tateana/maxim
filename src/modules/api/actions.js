import ArticleService from './ScoreService'
import NounService from './NounService'

export const FETCH_SERVICES = 'FETCH_SERVICES';
export const FETCH_SERVICES_SUCCESS = 'FETCH_SERVICES_SUCCESS';
export const FETCH_SERVICES_FAILED = 'FETCH_SERVICES_FAILED';


export const SAVE_ENTITY = 'SAVE_ENTITY'
export const SAVE_ENTITY_SUCCESS = 'SAVE_ENTITY_SUCCESS'
export const SAVE_ENTITY_FAILED = 'SAVE_ENTITY_FAILED'

export const FETCH_ARTICLE_SCORES = 'FETCH_ARTICLE_SCORES';
export const FETCH_SPELING_SCORES = 'FETCH_SPELING_SCORES';
export const FETCH_NOUNS = 'FETCH_SPELING_SCORES';


export const FETCH_WORD = 'FETCH_WORD';
export const FETCH_WORD_SUCCESS = 'FETCH_WORD_SUCCESS';
export const FETCH_WORD_NOT_FOUND = 'FETCH_WORD_NOT_FOUND';
export const FETCH_WORD_FAILED = 'FETCH_WORD_FAILED';


function fetchServices(services) {
    return {
        type: FETCH_SERVICES,
        payload: services
    }
}

export function fetchAll() {
    return fetchServices([ArticleService, NounService])
}

export function fetchArticles() {
    return fetchServices([ArticleService])
}

export function fetchNouns() {
    return fetchServices([NounService])
}

function saveEntity(item, service) {
    return {
        type: SAVE_ENTITY,
        payload: item,
        service
    }
}

export function saveArticleScore(item) {
    return saveEntity(item, ArticleService)
}

export function saveWord(item) {
    return saveEntity(item, NounService)
}

export function fetchDictItem(item) {
    return {
        type: FETCH_WORD,
        payload: item
    }
}