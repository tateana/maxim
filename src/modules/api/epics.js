import { catchError, mergeMap, map, switchMap } from 'rxjs/operators';
import { zip, of } from 'rxjs/index';
import { ofType } from 'redux-observable';
import * as a from './actions';
import ArticleService from './ScoreService'
import NounService from './NounService'
import YandexService from './yandexService'
import { epicErrorHandler } from '../utils'


export const fetchServices = action$ => action$.pipe(
    ofType(a.FETCH_SERVICES),
    switchMap(action => zip(...action.payload.map(service => service.fetchAll(1000))).pipe(
        map(result => ({
            type: a.FETCH_SERVICES_SUCCESS,
            payload: result
        })),
        catchError(error => of({
            type: a.FETCH_SERVICES_FAILED,
            payload: error,
            error: true
        }))
    )),
    epicErrorHandler
);


export const saveEntity = action$ => action$.pipe(
    ofType(a.SAVE_ENTITY),
    mergeMap(action => action.service.add(action.payload).pipe(
        map(() => ({
            type: a.SAVE_ENTITY_SUCCESS,
            payload: action.payload
        })),
        catchError(error => of({
            type: a.SAVE_ENTITY_FAILED,
            payload: error,
            error: true,
            entity: action.payload
        }))
    )),
    epicErrorHandler
);


export const fetchDictItem = action$ => action$.pipe(
    ofType(a.FETCH_WORD),
    switchMap(action => NounService.fetchExact(action.payload).pipe(
        switchMap(word => {
            if (word) {
                return zip(of(word), ArticleService.fetchExact(word.id))
            }

            return YandexService.find(action.payload)
        }),
        map(wordData => {
            if (!wordData) {
                return { type: a.FETCH_WORD_NOT_FOUND, payload: action.payload }
            }

            let payload = wordData
            if (!Array.isArray(wordData)) {
                payload = [wordData]
            }

            return {
                type: a.FETCH_WORD_SUCCESS,
                payload
            }
        }),
        catchError(errorObject => of({
            type: a.FETCH_WORD_FAILED,
            payload: errorObject.message,
            error: true
        })),
    )),
    epicErrorHandler
);
