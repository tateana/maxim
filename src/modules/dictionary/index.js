

import Dictionary from './Dictionary';

import * as actions from './actions';

export { actions }
export default Dictionary;