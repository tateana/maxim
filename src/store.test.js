
import { of } from 'rxjs/index';
import store from "./store";
import { Noun, Score } from './modules/api'
import { actions } from "./modules/dictionary"
import NounService from './modules/api/NounService'
import YandexService from './modules/api/yandexService'
import ArticleService from './modules/api/ScoreService'

ArticleService.db = null;
ArticleService.collection = null
NounService.db = null;
NounService.collection = null
NounService.fetchExact = jest.fn();
YandexService.find = jest.fn();
ArticleService.fetchExact = jest.fn();
NounService.add = jest.fn();
ArticleService.add = jest.fn();

const today = new Date('2018-11-20T18:53:31.633Z')

describe('dictionary flow dispatches the correct action and payload', () => {

    let triggeredActions = []
    const testSpec = [];
    let i = 0;

    store.attachReducers({
        testReducer: (state = null, action) => {
            triggeredActions.push(action)
            return state
        }
    })

    afterEach(() => {
        console.log(testSpec[i += 1].description, triggeredActions)
    });

    describe('Find Item', () => {
        beforeEach(() => {
            store.dispatch(actions.clear());
            triggeredActions = []
            NounService.fetchExact.mockClear()
            ArticleService.fetchExact.mockClear()
            YandexService.find.mockClear()
        })

        testSpec[1] = it('word and its score are found in database', () => {
            const noun = new Noun('Busch', null, 'm', 'куст', today)
            NounService.fetchExact.mockReturnValue(of(noun))
            const score = new Score('Busch', 2)
            ArticleService.fetchExact.mockReturnValue(of(score))

            store.dispatch(actions.findItem('busch'));
            expect(NounService.fetchExact).toHaveBeenCalledTimes(1);
            expect(ArticleService.fetchExact).toHaveBeenCalledTimes(1);
            expect(YandexService.find).not.toHaveBeenCalled();
            expect(triggeredActions).toMatchSnapshot();
        });

        testSpec[2] = it('word is not found in any source', () => {
            NounService.fetchExact.mockReturnValue(of(null))
            YandexService.find.mockReturnValue(of(false))

            store.dispatch(actions.findItem('Buch'));

            expect(NounService.fetchExact).toHaveBeenCalledTimes(1);
            expect(YandexService.find).toHaveBeenCalledTimes(1);
            expect(ArticleService.fetchExact).not.toHaveBeenCalled();
            expect(triggeredActions).toMatchSnapshot();
        });

        testSpec[3] = it('word is found in Yandex', () => {
            const noun = new Noun('Fenster', ['окно'], 'n', null, today)
            NounService.fetchExact.mockReturnValue(of(null))
            YandexService.find.mockReturnValue(of(noun))

            store.dispatch(actions.findItem('fenster'));

            expect(NounService.fetchExact).toHaveBeenCalledTimes(1);
            expect(YandexService.find).toHaveBeenCalledTimes(1);
            expect(ArticleService.fetchExact).not.toHaveBeenCalled();
            expect(triggeredActions).toMatchSnapshot();
        });
    });

    describe('Save Dict Items flow dispatches the correct action and payload', () => {


        beforeEach(() => {
            store.dispatch(actions.clear());
            triggeredActions = []
            NounService.add.mockClear()
            ArticleService.add.mockClear()
        })

        testSpec[4] = it('score saved successfully', () => {
            const noun = new Noun('Busch', null, 'm', 'куст', today)
            const score = new Score('Busch', 2)
            score.doModified()
            ArticleService.add.mockReturnValue(of(score))

            store.dispatch(actions.saveEntities([noun, score]));
            expect(ArticleService.add).toHaveBeenCalledTimes(1);
            expect(NounService.add).not.toHaveBeenCalled();
            expect(triggeredActions).toMatchSnapshot();
        });

        testSpec[5] = it('noun saved successfully', () => {
            const noun = new Noun('Tisch', null, 'm', 'стол', today)
            noun.doModified()
            const score = new Score('Tisch', 5)
            NounService.add.mockReturnValue(of(noun))

            store.dispatch(actions.saveEntities([noun, score]));
            expect(NounService.add).toHaveBeenCalledTimes(1);
            expect(ArticleService.add).not.toHaveBeenCalled();
            expect(triggeredActions).toMatchSnapshot();
        });

        testSpec[6] = it('noun and score are saved successfully', () => {
            const noun = new Noun('Tisch', null, 'm', 'стол', today)
            noun.doModified()
            const score = new Score('Tisch', 5)
            score.doModified()
            NounService.add.mockReturnValue(of(noun))
            ArticleService.add.mockReturnValue(of(score))

            store.dispatch(actions.saveEntities([noun, score]));
            expect(NounService.add).toHaveBeenCalledTimes(1);
            expect(ArticleService.add).toHaveBeenCalledTimes(1);
            expect(triggeredActions).toMatchSnapshot();
        });
    });
});