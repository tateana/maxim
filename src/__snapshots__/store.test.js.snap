// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dictionary flow dispatches the correct action and payload Find Item word and its score are found in database 1`] = `
Array [
  Object {
    "payload": "busch",
    "type": "DICTIONARY_ITEM_FIND",
  },
  Object {
    "payload": "busch",
    "type": "FETCH_WORD",
  },
  Object {
    "payload": Array [
      Noun {
        "_created": 2018-11-20T18:53:31.633Z,
        "_gender": "m",
        "_id": "Busch",
        "_isModified": false,
        "_ru": "куст",
        "_ruOptions": null,
      },
      Score {
        "_count": 2,
        "_id": "Busch",
        "_isModified": false,
        "_viewed": null,
      },
    ],
    "type": "FETCH_WORD_SUCCESS",
  },
  Object {
    "payload": Object {
      "Busch": Object {
        "articleScore": Score {
          "_count": 2,
          "_id": "Busch",
          "_isModified": false,
          "_viewed": null,
        },
        "word": Noun {
          "_created": 2018-11-20T18:53:31.633Z,
          "_gender": "m",
          "_id": "Busch",
          "_isModified": false,
          "_ru": "куст",
          "_ruOptions": null,
        },
      },
    },
    "type": "DICTIONARY_UPDATED",
  },
  Object {
    "payload": "Busch",
    "type": "DICTIONARY_ITEM_FOUND",
  },
]
`;

exports[`dictionary flow dispatches the correct action and payload Find Item word is found in Yandex 1`] = `
Array [
  Object {
    "payload": "fenster",
    "type": "DICTIONARY_ITEM_FIND",
  },
  Object {
    "payload": "fenster",
    "type": "FETCH_WORD",
  },
  Object {
    "payload": Array [
      Noun {
        "_created": 2018-11-20T18:53:31.633Z,
        "_gender": "n",
        "_id": "Fenster",
        "_isModified": false,
        "_ru": "окно",
        "_ruOptions": Array [
          "окно",
        ],
      },
    ],
    "type": "FETCH_WORD_SUCCESS",
  },
  Object {
    "payload": Object {
      "Fenster": Object {
        "word": Noun {
          "_created": 2018-11-20T18:53:31.633Z,
          "_gender": "n",
          "_id": "Fenster",
          "_isModified": false,
          "_ru": "окно",
          "_ruOptions": Array [
            "окно",
          ],
        },
      },
    },
    "type": "DICTIONARY_UPDATED",
  },
  Object {
    "payload": "Fenster",
    "type": "DICTIONARY_ITEM_FOUND",
  },
]
`;

exports[`dictionary flow dispatches the correct action and payload Find Item word is not found in any source 1`] = `
Array [
  Object {
    "payload": "Buch",
    "type": "DICTIONARY_ITEM_FIND",
  },
  Object {
    "payload": "Buch",
    "type": "FETCH_WORD",
  },
  Object {
    "payload": "Buch",
    "type": "FETCH_WORD_NOT_FOUND",
  },
  Object {
    "payload": false,
    "type": "DICTIONARY_ITEM_FOUND",
  },
]
`;

exports[`dictionary flow dispatches the correct action and payload Save Dict Items flow dispatches the correct action and payload noun and score are saved successfully 1`] = `
Array [
  Object {
    "payload": Array [
      Noun {
        "_created": 2018-11-20T18:53:31.633Z,
        "_gender": "m",
        "_id": "Tisch",
        "_isModified": false,
        "_ru": "стол",
        "_ruOptions": null,
      },
      Score {
        "_count": 5,
        "_id": "Tisch",
        "_isModified": false,
        "_viewed": null,
      },
    ],
    "type": "DICTIONARY_ITEMS_SAVE",
  },
  Object {
    "payload": Array [
      Noun {
        "_created": 2018-11-20T18:53:31.633Z,
        "_gender": "m",
        "_id": "Tisch",
        "_isModified": false,
        "_ru": "стол",
        "_ruOptions": null,
      },
      Score {
        "_count": 5,
        "_id": "Tisch",
        "_isModified": false,
        "_viewed": null,
      },
    ],
    "type": "DICTIONARY_UPDATE",
  },
  Object {
    "payload": Object {
      "Tisch": Object {
        "articleScore": Score {
          "_count": 5,
          "_id": "Tisch",
          "_isModified": false,
          "_viewed": null,
        },
        "word": Noun {
          "_created": 2018-11-20T18:53:31.633Z,
          "_gender": "m",
          "_id": "Tisch",
          "_isModified": false,
          "_ru": "стол",
          "_ruOptions": null,
        },
      },
    },
    "type": "DICTIONARY_UPDATED",
  },
  Object {
    "type": "DICTIONARY_SYNC",
  },
  Object {
    "payload": Noun {
      "_created": 2018-11-20T18:53:31.633Z,
      "_gender": "m",
      "_id": "Tisch",
      "_isModified": false,
      "_ru": "стол",
      "_ruOptions": null,
    },
    "service": NounService {
      "add": [Function],
      "collection": null,
      "db": null,
      "fetchExact": [Function],
    },
    "type": "SAVE_ENTITY",
  },
  Object {
    "payload": Score {
      "_count": 5,
      "_id": "Tisch",
      "_isModified": false,
      "_viewed": null,
    },
    "service": ScoreService {
      "add": [Function],
      "collection": null,
      "db": null,
      "fetchExact": [Function],
    },
    "type": "SAVE_ENTITY",
  },
  Object {
    "payload": Noun {
      "_created": 2018-11-20T18:53:31.633Z,
      "_gender": "m",
      "_id": "Tisch",
      "_isModified": false,
      "_ru": "стол",
      "_ruOptions": null,
    },
    "type": "SAVE_ENTITY_SUCCESS",
  },
  Object {
    "payload": Score {
      "_count": 5,
      "_id": "Tisch",
      "_isModified": false,
      "_viewed": null,
    },
    "type": "SAVE_ENTITY_SUCCESS",
  },
  Object {
    "payload": "Tisch",
    "type": "DICTIONARY_SYNCED",
  },
  Object {
    "type": "DICTIONARY_ITEMS_SAVED",
  },
]
`;

exports[`dictionary flow dispatches the correct action and payload Save Dict Items flow dispatches the correct action and payload noun saved successfully 1`] = `
Array [
  Object {
    "payload": Array [
      Noun {
        "_created": 2018-11-20T18:53:31.633Z,
        "_gender": "m",
        "_id": "Tisch",
        "_isModified": false,
        "_ru": "стол",
        "_ruOptions": null,
      },
      Score {
        "_count": 5,
        "_id": "Tisch",
        "_isModified": false,
        "_viewed": null,
      },
    ],
    "type": "DICTIONARY_ITEMS_SAVE",
  },
  Object {
    "payload": Array [
      Noun {
        "_created": 2018-11-20T18:53:31.633Z,
        "_gender": "m",
        "_id": "Tisch",
        "_isModified": false,
        "_ru": "стол",
        "_ruOptions": null,
      },
      Score {
        "_count": 5,
        "_id": "Tisch",
        "_isModified": false,
        "_viewed": null,
      },
    ],
    "type": "DICTIONARY_UPDATE",
  },
  Object {
    "payload": Object {
      "Tisch": Object {
        "articleScore": Score {
          "_count": 5,
          "_id": "Tisch",
          "_isModified": false,
          "_viewed": null,
        },
        "word": Noun {
          "_created": 2018-11-20T18:53:31.633Z,
          "_gender": "m",
          "_id": "Tisch",
          "_isModified": false,
          "_ru": "стол",
          "_ruOptions": null,
        },
      },
    },
    "type": "DICTIONARY_UPDATED",
  },
  Object {
    "type": "DICTIONARY_SYNC",
  },
  Object {
    "payload": Noun {
      "_created": 2018-11-20T18:53:31.633Z,
      "_gender": "m",
      "_id": "Tisch",
      "_isModified": false,
      "_ru": "стол",
      "_ruOptions": null,
    },
    "service": NounService {
      "add": [Function],
      "collection": null,
      "db": null,
      "fetchExact": [Function],
    },
    "type": "SAVE_ENTITY",
  },
  Object {
    "payload": Noun {
      "_created": 2018-11-20T18:53:31.633Z,
      "_gender": "m",
      "_id": "Tisch",
      "_isModified": false,
      "_ru": "стол",
      "_ruOptions": null,
    },
    "type": "SAVE_ENTITY_SUCCESS",
  },
  Object {
    "payload": "Tisch",
    "type": "DICTIONARY_SYNCED",
  },
  Object {
    "type": "DICTIONARY_ITEMS_SAVED",
  },
]
`;

exports[`dictionary flow dispatches the correct action and payload Save Dict Items flow dispatches the correct action and payload score saved successfully 1`] = `
Array [
  Object {
    "payload": Array [
      Noun {
        "_created": 2018-11-20T18:53:31.633Z,
        "_gender": "m",
        "_id": "Busch",
        "_isModified": false,
        "_ru": "куст",
        "_ruOptions": null,
      },
      Score {
        "_count": 2,
        "_id": "Busch",
        "_isModified": false,
        "_viewed": null,
      },
    ],
    "type": "DICTIONARY_ITEMS_SAVE",
  },
  Object {
    "payload": Array [
      Noun {
        "_created": 2018-11-20T18:53:31.633Z,
        "_gender": "m",
        "_id": "Busch",
        "_isModified": false,
        "_ru": "куст",
        "_ruOptions": null,
      },
      Score {
        "_count": 2,
        "_id": "Busch",
        "_isModified": false,
        "_viewed": null,
      },
    ],
    "type": "DICTIONARY_UPDATE",
  },
  Object {
    "payload": Object {
      "Busch": Object {
        "articleScore": Score {
          "_count": 2,
          "_id": "Busch",
          "_isModified": false,
          "_viewed": null,
        },
        "word": Noun {
          "_created": 2018-11-20T18:53:31.633Z,
          "_gender": "m",
          "_id": "Busch",
          "_isModified": false,
          "_ru": "куст",
          "_ruOptions": null,
        },
      },
    },
    "type": "DICTIONARY_UPDATED",
  },
  Object {
    "type": "DICTIONARY_SYNC",
  },
  Object {
    "payload": Score {
      "_count": 2,
      "_id": "Busch",
      "_isModified": false,
      "_viewed": null,
    },
    "service": ScoreService {
      "add": [Function],
      "collection": null,
      "db": null,
      "fetchExact": [Function],
    },
    "type": "SAVE_ENTITY",
  },
  Object {
    "payload": Score {
      "_count": 2,
      "_id": "Busch",
      "_isModified": false,
      "_viewed": null,
    },
    "type": "SAVE_ENTITY_SUCCESS",
  },
  Object {
    "payload": "Busch",
    "type": "DICTIONARY_SYNCED",
  },
  Object {
    "type": "DICTIONARY_ITEMS_SAVED",
  },
]
`;
